cli/build.go

```diff
@@ -250,3 +265,33 @@ func (o *BuildOptions) Run(ctx context.Context) error {
 	}
 	return nil
 }
+
+// AddDescriptors adds the attributes of each file listed in the config
+// to the annotations of its respective descriptor.
+func AddDescriptors(d []v1.Descriptor, c v1alpha1.DataSetConfiguration) []v1.Descriptor {
+	// For each descriptor
+	for i1, desc := range d {
+		// Get the filename of the block
+		filename := desc.Annotations["org.opencontainers.image.title"]
+		// For each file in the config
+		for i2, file := range c.Files {
+			// If the filename of the block matches the filename of the file in the config
+			// If the config has a grouping declared, make a valid regex.
+			if strings.Contains(file.File, "*") && !strings.Contains(file.File, ".*") {
+				file.File = strings.Replace(file.File, "*", ".*", -1)
+			}
+			namesearch := regexp.MustCompile(file.File)
```

Comment:
It would be more efficient to store the file information as a match and search by name when looping through the descriptions. Maybe something line `map[string]file.File`.